youtube_dl/extractor/medaltv.py

```diff
@@ -0,0 +1,127 @@
+# coding: utf-8
+
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import ExtractorError
+from ..compat import compat_str
+from ..utils import (
+    try_get,
+    float_or_none,
+    int_or_none
+)
+
+
+class MedalTVIE(InfoExtractor):
+    _VALID_URL = r'(?:https://)?(?:www\.)?medal\.tv(?:/clips)?/(?P<id>[0-9]+)'
```

Comment:
the `https?` part should not be optional.
do you have an example of clip URL without the `/clips` part?

Reply:
As mentioned in the pull request, I tested the extractor against 1300 medal clips. Some of these did not include the https, so I decided to make it optional. However, after seeing how other extractors work, I will make it required.

The choice to have `/clips` optional was also to do with another comment you made ( changing `url = 'https://medal.tv/clips/{}'.format(video_id)`). As some of those 1300 people submitted links like: `https://cdn.medal.tv/10619174/1080p-34934644.mp4`, and I thought it would be good to extract all formats from it (i.e. pretending it is `https://medal.tv/clips/10619174`. But it would seem as the default extractor is prioritized when dealing with direct links. I will change the regex to have `clips` as required.

Reply:
you can return the part without the `/clips/` but make it more restrictive(check for `cdn.medal.tv`) and add `matching_only` test.

Reply:
However, is it not the case that the default extractor is prioritized when dealing with direct links? Links from `cdn.medal.tv` are always direct links, so, regardless of the regex, it would use the default extractor?